ma/maInput.h

```diff
@@ -35,6 +35,9 @@ class Input
 {
   public:
     ~Input();
+    Input() {} // default empty c-tor
+    Input(const Input& in) = default; // copy c-tor
```

Comment:
Unless there's a good reason to keep it let's get rid of this copy since we have the move constructor which has more clear ownership semantics.

Reply:
@jacobmerson 

I am either missing something or this cannot be achieved by move semantics. It seems that this line

```
return new Input(std::move(*in));
```

in the current implementations of `makeAdvanced` actually uses the copy c-tor and not the move one (my best guess is because `in` is a constant). If I remove the copy c-tor I get a compile error 

```
error: use of deleted function ‘constexpr ma::Input::Input(const ma::Input&)’
```

Reply:
reverted things back to pre-move semantic state since the move does not seem to work here.